packages/ember-metal/lib/property_set.js

```diff
@@ -99,22 +99,17 @@ if (MANDATORY_SETTER) {
 }
 
 function setPath(root, path, value, tolerant) {
-  // get the last part of the path
-  let keyName = path.slice(path.lastIndexOf('.') + 1);
-
-  // get the first part of the part
-  path = (path === keyName) ? keyName : path.slice(0, path.length - (keyName.length + 1));
-
-  // unless the path is this, look up the first part to
-  // get the root
-  if (path !== 'this') {
```

Comment:
This is a very good catch, we removed support for this.paths, it was a workaround for capitalized paths back when they meant global (which never fully worked anyway).